presto-orc/src/test/java/com/facebook/presto/orc/TestOrcSelectiveStreamReaders.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.presto.orc;
+
+import com.facebook.presto.common.block.Block;
+import com.facebook.presto.common.type.Type;
+import com.facebook.presto.orc.metadata.CompressionKind;
+import com.facebook.presto.orc.reader.SelectiveStreamReader;
+import com.facebook.presto.orc.reader.SliceSelectiveStreamReader;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import org.testng.annotations.Test;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.stream.IntStream;
+
+import static com.facebook.presto.common.type.VarcharType.VARCHAR;
+import static com.facebook.presto.orc.OrcTester.Format.DWRF;
+import static com.facebook.presto.orc.OrcTester.assertBlockEquals;
+import static com.facebook.presto.orc.OrcTester.createCustomOrcSelectiveRecordReader;
+import static com.facebook.presto.orc.OrcTester.writeOrcColumnsPresto;
+import static com.facebook.presto.orc.TestingOrcPredicate.createOrcPredicate;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static com.google.common.collect.ImmutableMap.toImmutableMap;
+import static java.lang.Math.min;
+import static org.testng.Assert.assertEquals;
+
+public class TestOrcSelectiveStreamReaders
```

Comment:
This name is very generic, but the test itself applies only to SliceDirectSelectiveStreamReader. Perhaps, rename to TestSliceDirectSelectiveStreamReader.

Reply:
@mbasmanova I'm thinking to add more tests there, including more generic tests that cover all types, and also a test to resolve https://github.com/prestodb/presto/issues/15343 Do you think we can keep it as is for now?

Reply:
Sounds good.